### PR TITLE
Make the use of python's random saner and python 3.11 compatible

### PR DESCRIPTION
**Pull request by bp-kelley**
There were a few other changes along the way for both numpy and pickle.

### CODE REVIEW

File: rdkit/ML/Descriptors/UnitTestMolDescriptors.py [hunk 1]
```diff
@@ -92,6 +92,7 @@ def test_github3511(self):
         calculator.CalcDescriptors(mol)
 
         # This should not raise a pickling exception
+        print(mol.__dict__)
```
Comment (bp-kelley) [inline rdkit/ML/Descriptors/UnitTestMolDescriptors.py:95]: Remove

File: rdkit/ML/Descriptors/MoleculeDescriptors.py [hunk 2]
```diff
@@ -82,13 +82,15 @@ def CalcDescriptors(self, mol, *args, **kwargs):
...
     res = [-666] * len(self.simpleList)
+    mol.__getstate_manages_dict__ = False # this should be part of the pickle suite???
     for i, nm in enumerate(self.simpleList):
       fn = getattr(DescriptorsMod, nm, lambda x: 777)
       try:
         res[i] = fn(mol)
       except Exception:
         import traceback
         traceback.print_exc()
+    print(2, mol.__dict__)
```
Comment (bp-kelley) [inline rdkit/ML/Descriptors/MoleculeDescriptors.py:93]: Remove

File: rdkit/Chem/FeatMaps/FeatMapPoint.py [hunk 3]
```diff
@@ -14,7 +14,8 @@
 class FeatMapPoint(ChemicalFeatures.FreeChemicalFeature):
   weight = 0.0
   featDirs = None
-
+  __getstate_manages_dict__ = False
```
Comment (bp-kelley) [inline rdkit/Chem/FeatMaps/FeatMapPoint.py:17]: This is new...

File: rdkit/ML/Descriptors/MoleculeDescriptors.py [hunk 4]
```diff
@@ -82,13 +82,15 @@ def CalcDescriptors(self, mol, *args, **kwargs):
 
     """
     res = [-666] * len(self.simpleList)
+    mol.__getstate_manages_dict__ = False # this should be part of the pickle suite???
```
Comment (bp-kelley) [inline rdkit/ML/Descriptors/MoleculeDescriptors.py:85]: I'm curious why the .def_pickle doesn't handle this.

**Comment by greglandrum** [comment]
> I'm not sure why the doc tests are failing.

They are failing in the output from the BRICS tests, so I guess that's because the BRICS code (which is where the failures are happening) also makes heavy use of random.shuffle?

**Comment by bp-kelley** [comment]
Makes sense, I’ll check that.

**Comment by bp-kelley** [comment]
@greglandrum It's passing now.  Should I add a python 3.11 build?

**Comment by greglandrum** [comment]
I'm reluctant to add another build. Maybe we can replace one of the existing ones? 

Or maybe just a Linux 3.11 build?

I clearly have no strong opinion other than: let's not do one for each supported OS. 😀

**Comment by greglandrum** [comment]
@bp-kelley since this changes the results produced by multiple functions, it needs to be documented in the `Backwards incompatible changes` section of ReleaseNotes.md and should have something added to BackwardsIncompatibleChanges.md

**Comment by greglandrum** [comment]
@bp-kelley : I think it makes sense to close this now that #5994 has been merged. Agreed?

**Comment by bp-kelley** [comment]
Absolutely.

**Pull request closed without merging**